classes/class-newsmatchdonation-settings.php

```diff
@@ -0,0 +1,257 @@
+<?php
+/**
+ * The class for the News Match Donation plugin's setting
+ *
+ * @package NewsMatchDonation\Settings
+ */
+
+/**
+ * Register and populate a settings page for the News Match Donation Plugin:
+ */
+class NewsMatchDonation_Settings {
+	/**
+	 * The slug of the settings page
+	 *
+	 * @var string $settings_page The settings page slug
+	 */
+	public $settings_page = 'newsmatchdonation';
+
+	/**
+	 * The slug of the settings group
+	 *
+	 * @var string $settings_group The settings group slug
+	 */
+	public $settings_group = 'newsmatchdonation_group';
+
+	/**
+	 * The slug of the settings section
+	 *
+	 * @var string $settings_section The slug of the settings section
+	 */
+	public $settings_section = 'newsmatchdonation_section';
+
+	/**
+	 * The prefix used for this plugin's options saved in the options table
+	 *
+	 * @var string $options_prefix The prefix for this plugin's options saved in the options table
+	 */
+	public static $options_prefix = 'nmds_';
+
+	/**
+	 * Constructor for the settings class
+	 */
+	public function __construct() {
+		add_action( 'admin_menu', array( $this, 'register_submenu_page' ) );
+		add_action( 'admin_init', array( $this, 'register_settings' ) );
+	}
+
+	/**
+	 * Register the settings page
+	 */
+	public function register_submenu_page() {
+		add_submenu_page(
+			'plugins.php',
+			__( 'News Match Donation Shortcode', 'newsmatch' ),
```

Comment:
This should ideally contain an escaping function - I would recommend using [esc_html__()](https://developer.wordpress.org/reference/functions/esc_html__/) here